objects/obj_controller/Mouse_50.gml

```diff
@@ -112,96 +138,119 @@ if (menu=12) and (cooldown<=0) and (penitorium>0){
-        if (mouse_x>=xx+354) and (mouse_y>=yy+789) and (mouse_x<xx+448) and (mouse_y<yy+804) and (cooldown<=0){// Equip
-            var hue;hue=1;
-            if (obj_ini.artifact[menu_artifact]="Statue") then hue=0;
-            if (obj_ini.artifact[menu_artifact]="Casket") then hue=0;
-            if (obj_ini.artifact[menu_artifact]="Chalice") then hue=0;
-            if (obj_ini.artifact[menu_artifact]="Robot") then hue=0;
-            if (hue=1){
-                var pop;
-                pop=instance_create(0,0,obj_popup);
-                pop.type=8;cooldown=8;
-            }
-        }
-        if (mouse_x>=xx+512) and (mouse_y>=yy+789) and (mouse_x<xx+740) and (mouse_y<yy+804) and (cooldown<=0){// Gift
-            var pop,chick;chick=0;
-            if (known[2]>1) and (faction_defeated[2]=0) then chick=1;if (known[3]>1) and (faction_defeated[3]=0) then chick=1;
-            if (known[4]>1) and (faction_defeated[4]=0) then chick=1;if (known[5]>1) and (faction_defeated[5]=0) then chick=1;
-            if (known[6]>1) and (faction_defeated[6]=0) then chick=1;if (known[8]>1) and (faction_defeated[8]=0) then chick=1;
+    if (obj_ini.artifact_identified[menu_artifact]==0){
+        // Equip
+        if (mouse_x>=xx+354) and (mouse_y>=yy+789) and (mouse_x<xx+448) and (mouse_y<yy+804) and (cooldown<=0){
+            var hue=1;
+            if (obj_ini.artifact[menu_artifact]=="Statue") then hue=0;
+            if (obj_ini.artifact[menu_artifact]=="Casket") then hue=0;
+            if (obj_ini.artifact[menu_artifact]=="Chalice") then hue=0;
+            if (obj_ini.artifact[menu_artifact]=="Robot") then hue=0;
+            if (hue==1){
+                var pop=instance_create(0,0,obj_popup);
+                pop.type=8;
+                cooldown=8;
+            }
+        }
+        // Gift
+        if (mouse_x>=xx+512) and (mouse_y>=yy+789) and (mouse_x<xx+740) and (mouse_y<yy+804) and (cooldown<=0){
+            var chick=0;
+            if (known[2]>1) and (faction_defeated[2]==0) then chick=1;
+            if (known[3]>1) and (faction_defeated[3]==0) then chick=1;
+            if (known[4]>1) and (faction_defeated[4]==0) then chick=1;
+            if (known[5]>1) and (faction_defeated[5]==0) then chick=1;
+            if (known[6]>1) and (faction_defeated[6]==0) then chick=1;
+            if (known[8]>1) and (faction_defeated[8]==0) then chick=1;
             if (chick!=0){
-                pop=instance_create(0,0,obj_popup);
-                pop.type=9;cooldown=8;
+                var pop=instance_create(0,0,obj_popup);
+                pop.type=9;
+                cooldown=8;
             }
         }
-        if (mouse_x>=xx+780) and (mouse_y>=yy+789) and (mouse_x<xx+894) and (mouse_y<yy+804) and (cooldown<=0){// Destroy
-            var fun;fun=floor(random(100))+1;
+        // Destroy
+        if (mouse_x>=xx+780) and (mouse_y>=yy+789) and (mouse_x<xx+894) and (mouse_y<yy+804) and (cooldown<=0){
```

Comment:
in general i think we should use the point_in_rectangle() function gamemaker ships with for these